File: BinarySearch/C++/BinarySearch.cpp
#include <iostream>

using namespace std;

void BinSearch(int arr[], int number, int N)
{
	int low = 0;                                       // ãðàíèöû ðàññìàòðèâàåìîãî ïðîìåæóòêà ìàññèâà 
  	int high = N;

  	while(low < high - 1) {                            // ïîêà ñóùåñòâóåò ðàññìàòðèâàåìûé ïðîìåæóòîê
    	int mid = (low + high) / 2;                    // èíäåêñ ñðåäíåãî ýëåìåíòà ðàññìàòðèâàåìîãî ïðîìåæóòêà ìàññèâà
    	
    	if(number < arr[mid])                          // ñóæàåì ãðàíèöû ðàññìàòðèâàåìîãî ïðîìåæóòêà ìàññèâà
		{
      		high = mid;
    	} 
		else 
		{
      		low = mid;
    	}
  	}
   	
  	if(arr[low] == number) {                           // åñëè number íàéäåíà, òî âûâîäèì èíäåêñ
    	cout << "arr[" << low << "] = " << number;
  	} 
  	else 
  	{
    	cout << number << " - not found";
  	}
}

int main() {
  	int N, number; 
	cin >> N;                                          // ââîäèì ÷èñëî ýëåìåíòîâ ìàññèâà

	int arr[N];

	for(int i = 0; i < N; i++) {                       // ââîäèì ìàññèâ
		cin >> arr[i];
	}

	cin >> number;                                     // ÷èñëî, êîòîðîå íåîáõîäèìî íàéòè â ìàññèâå

	BinSearch(arr, number, N);
}

File: BinarySearch/Java/BinarySearch.java
package com.company;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
    int number = 0;
    int n;

	Scanner scan = new Scanner(System.in);

	n = scan.nextInt();

	int arr [] = new int[n];

	for(int i = 0; i < n; i++){
		int item = scan.nextInt();
		arr[i] = item;
	}

	System.out.println();

	for(int i = 0; i < n; i++){
		System.out.println(arr[i]);
	}

	number = scan.nextInt();

	int low = 0;
	int high = n;

	while(high - 1 > low){
		int mid = (low + high) / 2;

		if(number < arr[mid]){
			high = mid;
		}
		else {
			low = mid;
		}
 	}

	if(arr[low] == number){
		System.out.println(low);
	}
	else {
		System.out.println("Not found");
	}
    }
}
